### PR TITLE
minusDiv -> getLerpProgress + other improvements

### PR DESCRIPTION
**Pull request by Earthcomputer**
Someone didn't look at this function very carefully :P

### CODE REVIEW

File: mappings/net/minecraft/util/math/MathHelper.mapping [hunk 1]
```diff
@@ -76,10 +76,15 @@ CLASS net/minecraft/class_3532 net/minecraft/util/math/MathHelper
...
-		ARG 0 numerator
-		ARG 2 delta
-		ARG 4 denominator
+	METHOD method_15370 inverseLerp (DDD)D
+		COMMENT Essentially does the inverse of the linear interpolation (lerp) function.
+		COMMENT Returns the delta value needed to interpolate between {@code a} and {@code b}.
+		ARG 0 value
+			COMMENT The result of the lerp function
+		ARG 2 a
+			COMMENT The value interpolated from
```
Comment (liach) [inline mappings/net/minecraft/util/math/MathHelper.mapping:88]: Maybe specify this is the lower bound (smaller)?

File: mappings/net/minecraft/util/math/MathHelper.mapping [hunk 2]
```diff
@@ -76,10 +76,15 @@ CLASS net/minecraft/class_3532 net/minecraft/util/math/MathHelper
 		ARG 0 hue
 		ARG 1 saturation
 		ARG 2 value
-	METHOD method_15370 minusDiv (DDD)D
-		ARG 0 numerator
-		ARG 2 delta
-		ARG 4 denominator
+	METHOD method_15370 inverseLerp (DDD)D
+		COMMENT Essentially does the inverse of the linear interpolation (lerp) function.
```
Comment (liach) [inline mappings/net/minecraft/util/math/MathHelper.mapping]: Maybe say given delta, a, b, `inverseLerp(lerp(delta, a, b), a, b) == delta`? I thought it's like lerping the inversed `value` at first sight.

**Review by liach — COMMENTED** [review]
Sorry, but it's me who [named it `minusDiv`](https://github.com/FabricMC/yarn/commit/94b94160eafb7d46518dfd18c114950afb58df68#diff-d8865f4320c6cc9692e2b0048a63712eR90-R93).

**Comment by Runemoro** [comment]
Is `inverseLerp` the best name for this? Maybe something like `getProgress(value, start, end)` would be a better name, since it's calculating the "progress" from the start value to the end value given the current value.

**Comment by liach** [comment]
```java
            this.lastNetherPortalPosition = new BlockPos(pos);
            NetherPortalBlock class_2423 = (NetherPortalBlock)Blocks.NETHER_PORTAL;
            BlockPattern.Result result3 = NetherPortalBlock.findPortal(this.world, this.lastNetherPortalPosition);
            double double4 = (result3.getForwards().getAxis() == Direction.Axis.X) ? result3.getFrontTopLeft().getZ() : ((double)result3.getFrontTopLeft().getX());
            double double6 = Math.abs(MathHelper.minusDiv(((result3.getForwards().getAxis() == Direction.Axis.X) ? this.getZ() : this.getX()) - (double)((result3.getForwards().rotateYClockwise().getDirection() == Direction.AxisDirection.NEGATIVE) ? 1 : 0), double4, double4 - result3.getWidth()));
            double double8 = MathHelper.minusDiv(this.getY() - 1.0, result3.getFrontTopLeft().getY(), result3.getFrontTopLeft().getY() - result3.getHeight());
            this.lastNetherPortalDirectionVector = new Vec3d(double6, double8, 0.0);
            this.lastNetherPortalDirection = result3.getForwards();
```
Fyi this is like the only place where this method is used (in entity's portal code).

**Comment by Earthcomputer** [comment]
inverseLerp is definitely a better name than getProgress, especially since this is the math class. However I approve of the start and end parameter names.

**Comment by Runemoro** [comment]
> inverseLerp is definitely a better name than getProgress

The term "progress" is used in many other places in yarn to refer to this kind of value. I don't see why you think the method should be described as the "inverse of lerp" when it can be directly described as "progress". It would be a bit like calling division "inverse multiplication".

The javadocs also focus on this being "the delta value needed to interpolate between", even though its only usage isn't to find the delta for an interpolation.

> especially since this is the math class

I don't see what you mean by that.

**Comment by Earthcomputer** [comment]
After some more thought and discussion on discord, I have decided to name the function `getLerpProgress`. `getProgress` on its own is too vague, whereas `inverseLerp` is a little difficult to read, especially given that it only inverts for one variable.
I will also update the javadocs to describe the function first without using lerp, then later explain its link to lerp.

**Comment by Earthcomputer** [comment]
I also made a couple of other improvements to MathHelper.

**Comment by liach** [comment]
Can you rename `randomUUID` to `randomUuid` as well @Earthcomputer?

**Comment by Earthcomputer** [comment]
Done

File: mappings/net/minecraft/util/math/MathHelper.mapping [hunk 3]
```diff
@@ -156,6 +195,7 @@ CLASS net/minecraft/class_3532 net/minecraft/util/math/MathHelper
 		ARG 0 start
 		ARG 1 end
 		ARG 2 delta
+	METHOD method_22860 fwrapDegrees (D)F
```
Comment (liach) [inline mappings/net/minecraft/util/math/MathHelper.mapping:198]: May be `fWrapDegrees` or `wrapDegreesF` (note your casing)

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/util/math/MathHelper.mapping (hunk 3 above)
Comment (Earthcomputer) [reply mappings/net/minecraft/util/math/MathHelper.mapping:198]: I was following the pattern of `lfloor`, it differs from `wrapDegrees(D)D` only in return type. Also, this is a deprecated method that people should probably use the other methods in favour of.

**Review by Juuxel — APPROVED** [review]

File: mappings/net/minecraft/util/math/MathHelper.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/util/math/MathHelper.mapping:198]: Not really, we have `nextInt` `nextDouble`, so `floorLong` etc should be more accurate and match the pattern better

**Comment by liach** [comment]
Can be merged once branch is 20w09a

**Comment by Earthcomputer** [comment]
Done

**Pull request merged**